game/src/Game/Autoload/CursorManager.gd

```diff
@@ -0,0 +1,173 @@
+extends Node
+
+class CompatCursor:
+	#cursor properties
+	var cursor_name : String
+
+	var resolutions : PackedVector2Array
+	var frames : Array[ImageTexture]
+	var hotspots : PackedVector2Array
+	var is_animated : bool = false
+	var sequence : PackedInt32Array = [0]
+	var timings : PackedFloat32Array = [1.0]
+	
+	#Cursor state
+	var currentFrame : int = 0
+	var timeToFrame : float = 1.0
+	
+	func _init(nameIn:String) -> void:
+		self.cursor_name = nameIn
+		self.resolutions = CursorSingleton.get_resolutions(self.cursor_name)
+
+		self.frames = CursorSingleton.get_frames(self.cursor_name,0)
+		self.hotspots = CursorSingleton.get_hotspots(self.cursor_name,0)
+
+		self.is_animated = len(frames) > 1
+		if self.is_animated:
+			self.sequence = CursorSingleton.get_sequence(self.cursor_name)
+			self.timings = CursorSingleton.get_displayRates(self.cursor_name)
+			self.timeToFrame = self.timings[self.sequence[currentFrame]]
+	
+	func reset() -> void:
+		currentFrame = 0
+		timeToFrame = self.timings[self.sequence[0]]
+	
+	func set_resolution(resolution : Vector2) -> void:
+		var index : int = resolutions.find(resolution)
+		if index != -1:
+			frames = CursorSingleton.get_frames(cursor_name,index)
+			return
+
+		#couldnt find it, so generate it based on the highest res available
+		var highest_res_index : int = 0
+		var highest_res_x : int = 0
+		for i : int in range(len(resolutions)):
+			if resolutions[i].x > highest_res_x:
+				highest_res_x = resolutions[i].x
+				highest_res_index = i
+		generate_new_res(highest_res_index,resolution)
+		
+		self.resolutions = CursorSingleton.get_resolutions(self.cursor_name)
+		self.frames = CursorSingleton.get_frames(self.cursor_name,len(self.resolutions)-1)
+		self.hotspots = CursorSingleton.get_hotspots(self.cursor_name,len(self.resolutions)-1)
+		
+		assert(len(self.frames) != 0)
+
+	func generate_new_res(base_res_index:int, resolution:Vector2) -> void:
+		# resolution wasn't in among the default, need to generate it ourselves
+		CursorSingleton.generate_resolution(cursor_name,base_res_index,resolution)
+
+	#only bother with this if the cursor is animated
+	func _process_cursor(delta:float, shape:Input.CursorShape=Input.CURSOR_ARROW) -> void:
+		timeToFrame -= delta
+		if(timeToFrame <= 0):
+			currentFrame = (currentFrame + 1) % len(sequence)
+			timeToFrame = timings[self.sequence[currentFrame]] 
```

Comment:
Maybe this should be `+=` rather than `=` so we don't ignore whatever portion of `delta` is still left in `timeToFrame`? I guess if there's significant lag then the next couple calls of `_process_cursor` will all trigger a frame change, but that should catch up quickly (we could change the `if` to a `while` and ensure we move to the right frame straight away, maybe holding off on calling `set_hardware_cursor` until the end (assuming we went through the loop at least once) to avoid redundant calls, but tbh this might be overkill for a simple cursor animation function).